includes/gateways/cod/class-wc-gateway-cod.php

```diff
@@ -180,8 +180,6 @@ public function is_available() {
 					}
 				}
 			}
-		} elseif ( WC()->cart && WC()->cart->needs_shipping() ) {
-			$needs_shipping = true;
```

Comment:
This will make `WC()->cart && WC()->cart->needs_shipping()` have `$needs_shipping` equal to `false`.

Reply:
@claudiosanches We are checking that twice. I just removed the bottom one. :smiley: . Please have a look at the screenshot-
![screenshot-2018-06-11-14 53 38-001](https://user-images.githubusercontent.com/6243408/41222083-d78b9f82-6d87-11e8-9bf3-a162eb29bfda.png)

First `if` is in [line 167](https://github.com/rnaby/woocommerce/blob/ec76e60e33911050c67601cbbf496c36dca0faaf/includes/gateways/cod/class-wc-gateway-cod.php#L167) and the changed `elseif` is in [line 183](https://github.com/rnaby/woocommerce/blob/ec76e60e33911050c67601cbbf496c36dca0faaf/includes/gateways/cod/class-wc-gateway-cod.php#L183).